cmd/manager/manager.go

```diff
@@ -96,6 +111,15 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Barrier to ensure that the operator is not started on Kubernetes lower than 1.21.
+	// This is to avoid the issue that the operator will be stuck in a crash loop if it is started on Kubernetes lower than 1.21.
+	kubernetesVersion, err := discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig()).ServerVersion()
+	if err != nil {
+		setupLog.Error(err, "Unable to get Kubernetes version")
+		os.Exit(1)
+	}
+	requireKubernetesVersion(kubernetesVersion, 1, 21)
```

Comment:
Do we want to hardened this? Fine by me, I just don't know how often this would change.

Reply:
Not very often I believe.